### PR TITLE
Adds full text search relevance helper.

### PR DESCRIPTION
**Pull request by angel-rec-ct**

### CODE REVIEW

File: src/Utility/Models/Criteriable.php [hunk 1]
```diff
@@ -22,6 +22,11 @@ trait Criteriable
     protected ?array $sortables = null;
     protected string $fullTextSearchMode = '';
 
+    /**
+     * @var callable(Builder):int $relevanceMethod
+     */
+    protected $relevanceMethod;
```
Comment (mshamaseen) [inline src/Utility/Models/Criteriable.php]: it seems that this property is not used anywhere?

File: src/Utility/Models/Criteriable.php [hunk 2]
```diff
@@ -53,43 +58,52 @@ public function scopeFilterByCriteria($query, array $criteria): Builder
...
-                        });
-                    } else {
-                        $this->buildMatchAgainst($q, $columns, $criteria['search']);
-                    }
+        if (!isset($criteria['search'])) {
+            return $query;
+        }
+
+        if($relevance && is_callable($relevance)) {
+            $relevance($query, $criteria);
```
Comment (mshamaseen) [inline src/Utility/Models/Criteriable.php]: sorry but this doesn't make sense, the `searchByCriteria` method is already called in the builder, in the developer needs to add more query he can add them on the builder chain, they don't need to add a query as a parameter in this method. am I missing something?

**Review by mshamaseen — CHANGES_REQUESTED** [review]

File: src/Utility/Models/Criteriable.php (hunk 1 above)
Comment (angel-rec-ct) [reply src/Utility/Models/Criteriable.php]: yes, removing it

File: src/Utility/Models/Criteriable.php (hunk 2 above)
Comment (angel-rec-ct) [reply src/Utility/Models/Criteriable.php]: Then in that case there's no need for changes to make the relevance work inside the package, just let me update the docs

**Pull request merged**